### DISCUSSION TITLE
Role(s)

### DISCUSSION THREAD

**Discussion by jeisenbath** [jeisenbath/ansible-collection-solarwinds-orion#10] [Ideas]
Hey all,

I plan to work on a Role or two to add the the collection in the next release. I haven't quite decided how to split them up, whether by polling type (i.e. icmp, snmp, wmi node), or by device type (i.e. linux server, windows server, cisco router). Would love to hear feedback on how you may use these roles, if at all.

Thanks!

**Comment by Andyjb8** [comment]
Device type would make sense to me. I have some nodes that are just ICMP, but mostly have different flavors of Cisco/Juniper switches, routers, firewalls that all use snmp, plus linux servers with snmp.

**Reply by jeisenbath** [reply]
I think I'm leaning towards this too, because custom pollers using imported MIBs and custom OIDs would fit more into a device type role.

**Comment by jeisenbath** [comment]
New release dropped with a very generic role, which I settled on after working heavily with Dynamic inventories. I believe now that roles should be as widely applicable as possible, with the ability to control behavior with playbook, group, or host variables. If anyone has questions on use of the role, please drop a comment here!
